### PR TITLE
[EdgeTpu] Separate ConfigObject's logic dependent on Backend by introducing ConfigSetting

### PR DESCRIPTION
**Pull request by imsw0529**
Related to issue #1621, draft #1639

This commit separate ConfigObject's logic dependent on Backend by introducing ConfigSetting

- Introduce abstract class ConfigSetting for separate logic
- Introduce class OneConfigSetting extends ConfigSetting for ONE Backend logic
- Apply OneConfigSetting to ConfigObject
- Add Test code for ConfigSetting associated code

Co-author: SeungHyunH <vkflek1232@naver.com>

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

### CODE REVIEW

**Comment by SeungHyunH** [comment]
LGTM :D

File: src/OneExplorer/ConfigObject.ts [hunk 1]
```diff
@@ -112,12 +121,31 @@ export class ConfigObj {
...
+
+    return configSetting;
+  }
+
   private constructor(uri: vscode.Uri, rawObj: Cfg) {
     this.uri = uri;
     this.rawObj = rawObj;
+    this.configType = CfgType.one;
+
+    // separate to init()
```
Comment (dayo09) [inline src/OneExplorer/ConfigObject.ts:144]:
```suggestion
    // TODO: separate to init()
```

Comment (dayo09) [reply src/OneExplorer/ConfigObject.ts:144]: Could you just leave this `TODO` together?

Comment (imsw0529) [reply src/OneExplorer/ConfigObject.ts]: Thank you! I will add `TODO` comment.

**Review by dayo09 — APPROVED** [review]
LGTM, Thanks! :smile:

**Review by SeungHyunH — COMMENTED** [review]
LGTM

**Review by SeungHyunH — APPROVED** [review]

**Pull request merged**